core/src/utils/internal/dom.ts

```diff
@@ -87,3 +87,11 @@ export function addEvent(element: Element, type: string, fn: EventListenerOrEven
 		element.removeEventListener(type, fn);
 	};
 }
+
+let idCount = 0;
+/**
+ * Generates a unique ID with the format 'auId-[counter]'.
+ *
+ * @returns The generated ID.
+ */
+export const getGeneratedId = () => `auId-${idCount++}`;
```

Comment:
maybe rename to **generateId** ?
I usually prefer to add the *get* prefix only when the rest of the name does not clarify that it returns something.

another thing: after merging this, create an issue to use this in the accordion item, which also manages a unique id.